File: scripts/translate-prompts.js
// 翻译脚本 - 将英文提示词翻译为中文
const fs = require('fs');

// 简单的翻译函数 - 实际项目中应使用专业翻译API
// 这里我们使用占位符，实际使用时替换为真正的翻译调用
function translateToChinese(text) {
  if (!text || typeof text !== 'string') return text;

  // 对于JSON对象，需要翻译所有字符串值但保留键
  if (text.startsWith('{')) {
    try {
      const obj = JSON.parse(text);
      const translated = translateObject(obj);
      return JSON.stringify(translated, null, 2);
    } catch (e) {
      return text;
    }
  }

  // 对于普通文本，返回原文（实际翻译时替换为API调用）
  return text;
}

// 递归翻译对象中的所有字符串值
function translateObject(obj) {
  if (typeof obj === 'string') {
    return obj; // 翻译时替换为 translateApi(obj)
  }

  if (Array.isArray(obj)) {
    return obj.map(item => translateObject(item));
  }

  if (obj && typeof obj === 'object') {
    const result = {};
    for (const key in obj) {
      result[key] = translateObject(obj[key]);
    }
    return result;
  }

  return obj;
}

// 主处理函数
function processFile() {
  const filePath = 'src/data/resource-library/categories/prompts-chat-image.json';
  const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));

  console.log(`开始处理 ${data.prompts.length} 个提示词...`);

  // 先检查现有结构
  const sample = data.prompts[0];
  console.log('字段检查:');
  console.log('- nameEn 存在:', !!sample.nameEn);
  console.log('- contentEn 存在:', !!sample.contentEn);
  console.log('- descriptionEn 存在:', !!sample.descriptionEn);
  console.log('- name 存在:', !!sample.name);
  console.log('- content 存在:', !!sample.content);
  console.log('- description 存在:', !!sample.description);

  // 分析需要翻译的内容
  let totalContentSize = 0;
  data.prompts.forEach((p, i) => {
    if (p.contentEn) {
      totalContentSize += p.contentEn.length;
    }
  });
  console.log(`\n总contentEn大小: ${(totalContentSize / 1024).toFixed(1)} KB`);

  console.log('\n脚本就绪。实际翻译需要调用翻译API。');
}

processFile();

File: scripts/translate_prompts.py
#!/usr/bin/env python3
"""
翻译脚本 - 将英文提示词翻译为中文
"""
import json
import sys
import time
import traceback
from deep_translator import GoogleTranslator

# 初始化翻译器
translator = GoogleTranslator(source='en', target='zh-CN')

def translate_text(text, max_retries=3):
    """翻译文本，带重试机制"""
    if not text or not isinstance(text, str):
        return text

    # 如果是JSON字符串，尝试解析并翻译
    if text.strip().startswith('{'):
        try:
            obj = json.loads(text)
            translated_obj = translate_object(obj)
            return json.dumps(translated_obj, ensure_ascii=False, indent=2)
        except Exception as e:
            print(f"  JSON解析错误: {e}", file=sys.stderr)

    # 如果文本太长，分段翻译
    max_len = 4000
    if len(text) > max_len:
        paragraphs = text.split('\n')
        result = []
        current = ""
        for para in paragraphs:
            if len(current) + len(para) + 1 > max_len:
                if current:
                    try:
                        result.append(translator.translate(current))
                    except Exception as e:
                        print(f"  翻译错误: {e}", file=sys.stderr)
                        result.append(current)
                    time.sleep(0.3)
                current = para
            else:
                current = current + "\n" + para if current else para
        if current:
            try:
                result.append(translator.translate(current))
            except:
                result.append(current)
        return '\n'.join(result)

    for attempt in range(max_retries):
        try:
            return translator.translate(text)
        except Exception as e:
            if attempt < max_retries - 1:
                time.sleep(1)
            else:
                print(f"  翻译失败: {e}", file=sys.stderr)
                return text
    return text

def translate_object(obj):
    """递归翻译对象中的字符串值"""
    if isinstance(obj, str):
        return translate_text(obj)
    elif isinstance(obj, list):
        return [translate_object(item) for item in obj]
    elif isinstance(obj, dict):
        result = {}
        for key, value in obj.items():
            result[key] = translate_object(value)
        return result
    return obj

def main():
    input_file = 'src/data/resource-library/categories/prompts-chat-image.json'
    output_file = 'src/data/resource-library/categories/prompts-chat-image.json'

    print("读取文件...")
    sys.stdout.flush()

    with open(input_file, 'r', encoding='utf-8') as f:
        data = json.load(f)

    prompts = data['prompts']
    total = len(prompts)
    print(f"共 {total} 个提示词需要翻译")
    sys.stdout.flush()

    for i, prompt in enumerate(prompts):
        name = prompt.get('nameEn', '')[:60]
        print(f"[{i+1}/{total}] 翻译: {name}...", end=" ", flush=True)

        try:
            # 翻译 nameEn -> name
            if 'nameEn' in prompt and prompt['nameEn']:
                prompt['name'] = translate_text(prompt['nameEn'])
                time.sleep(0.2)

            # 翻译 descriptionEn -> description
            if 'descriptionEn' in prompt and prompt['descriptionEn']:
                prompt['description'] = translate_text(prompt['descriptionEn'])
                time.sleep(0.2)

            # 翻译 contentEn -> content
            if 'contentEn' in prompt and prompt['contentEn']:
                prompt['content'] = translate_text(prompt['contentEn'])
                time.sleep(0.3)

            print("完成")
            sys.stdout.flush()

        except Exception as e:
            print(f"错误: {e}")
            traceback.print_exc()

        # 每处理20个保存一次进度
        if (i + 1) % 20 == 0:
            print(f"  保存进度 ({i+1}/{total})...")
            sys.stdout.flush()
            with open(output_file, 'w', encoding='utf-8') as f:
                json.dump(data, f, ensure_ascii=False, indent=2)

    # 保存最终结果
    print("保存最终结果...")
    with open(output_file, 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)

    print("完成!")

if __name__ == '__main__':
    main()
